addons/purchase/models/mail_compose_message.py

```diff
@@ -8,7 +8,7 @@ class MailComposeMessage(models.TransientModel):
     _inherit = 'mail.compose.message'
 
     def _action_send_mail(self, auto_commit=False):
-        if self.model == 'purchase.order':
+        if 'purchase.order' in self.mapped('model'):
```

Comment:
I think you should only consider records whose model is the one you're expecting, e.g.:

```suggestion
        for compose in self.filtered(lambda c: c.model == 'purchase.order'):
```

Same in similar cases.

Reply:
agree

Reply:
@luisg123v 

how we could handle the `return super..`?  you propose a for loop, but it's needed the return in a single call
so the mapped that made @luistorresm seems to be correct

Reply:
@hugho-ad 

> so the mapped that made @luistorresm seems to be correct

The problem with that is it assumes::
- All composers belong to the same model
- All composers are for the same partner, context changes according to partners

It would need to be something like:

```
same_model = self.filtered(lambda c: c.model == 'purchase.order')
other_model = self - same_model
result_mails_su, result_messages = super(MailComposeMessage, other_model)._action_send_mail(auto_commit=auto_commit)
for compose in same_model.with_context(mailing_document_based=True)
    if self.env.context.get('mark_rfq_as_sent'):
        compose = compose.with_context(mail_notify_author=self.env.user.partner_id in compose.partner_ids)
    mail, message = super(MailComposeMessage, compose)._action_send_mail(auto_commit=auto_commit)
    result_mails_su |= mail
    result_messages |= message
return result_mails_su, result_messages
```

Reply:
Indeed, I believe only two approaches would be acceptable here:

1) calling super one record at a time, as suggested by Luis above
2) only try to modify the context of the super call when `len(self) == 1 or len(self.mapped('model') == 1`. 
Considering there is no standard code calling the method with a recordset of multiple wizards, I'm not sure we need to handle the context change when the method is called with wizards targeting different models.

Reply:
Done @Feyensv @luisg123v 

I prefer the second option to avoid many changes in the methods.

Regards